include/seqan3/argument_parser/argument_parser.hpp

```diff
@@ -322,6 +337,92 @@ class argument_parser
         if (parse_was_called)
             throw parser_design_error("The function parse() must only be called once!");
 
+        if (version_check) // was not disabled at construction
+        {
+            detail::version_checker app_version{info.app_name, info.version};
+
+            if (!version_check_option.has_value()) // the user did not specify the --version-check option
+            {
+                if (std::getenv("SEQAN3_NO_VERSION_CHECK") != nullptr) // environment variable was set
+                {
+                    version_check_option = false;
+                    debug_stream << "nop"<< std::endl;
+                }
+                else
+                {
+                    if (detail::is_terminal())
+                    {
+                        std::cerr << R"(
+#######################################################################
+   Automatic Update Notifications
+#######################################################################
+
+ Applications built with SeqAn can check for updates automatically.
+ This ensures you will be informed about new versions and important
+ fixes and is highly recommended. It involves making a web-request
+ to a server and transmitting basic application metadata (of course
+ not your data files!). For privacy implications, please see:
+ https://github.com/seqan/seqan3/wiki/Update-Notifications
+
+    [a] Always perform version checks for this app (the default).
+    [n] Never perform version checks for this app.
+    [y] Yes, perform a version check now, and ask again next time.
+    [s] Skip the version check now, but ask again next time.
+
+ You can manually adjust the behaviour of a single program call by adding option
+      --version-check 1/0 (version check on/off)
+
+ or you can permanently disable the version check by setting the environment
+ variable SEQAN3_NO_VERSION_CHECK.
+
+ Please enter one of [a, n, y, s]
+#######################################################################
```

Comment:
maybe actually we should make the whole thing shorter... what do you think about:
```
#######################################################################
   Automatic Update Notifications
#######################################################################

 This app can look for updates automatically in the background, 
 do you want to do that?

    [a] Always perform version checks for this app (the default).
    [n] Never perform version checks for this app.
    [y] Yes, perform a version check now, and ask again next time.
    [s] Skip the version check now, but ask again next time.

 Please enter one of [a, n, y, s] and press [RETURN].

 For more information, see:
 https://github.com/seqan/seqan3/wiki/Update-Notifications

#######################################################################